src/app/misc/index.ts

```diff
@@ -6,11 +6,12 @@ export const urlValidator: ValidatorFn = (control: AbstractControl) => {
   }
 
   const v: string = control.value;
-  return /^(?:(?:(?:https?|ftp):)?\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})).?)(?::\d{2,5})?(?:[/?#]\S*)?$/i.test(
-    v
-  )
-    ? null
-    : { url: true };
+  try {
+    new URL(v);
+    return { url: true };
+  } catch (err) {
+    return null;
```

Comment:
this should be the other way around. If URL is created it should return `null`. The returned object is represents the error

Reply:
got it, updating